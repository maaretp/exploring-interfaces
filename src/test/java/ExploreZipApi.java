import static io.restassured.RestAssured.given;

import org.approvaltests.Approvals;

import io.restassured.response.Response;
import junit.framework.TestCase;

public class ExploreZipApi extends TestCase {

		public void testZip() throws Exception {
			 Response response = given().get("https://api.zippopotam.us/JE/JE3");
			 Approvals.verify("status code: " + response.getStatusCode() 
			 			+ "\n" + "headers: \n"
			 			+ response.headers().toString()
		 				+ "\n" + "body: \n" 
		 				+ response.body().prettyPrint());
		}
		
		public void	testMaskedZip() throws Exception {
			Approvals.verify(getMaskedHeader());
		}
		
		
		
		 public String getMaskedHeader(){
		     Response response = given().get("http://api.zippopotam.us/us/20500"); 
			 String responseString = response.headers().toString();
			 
				 responseString = responseString.replaceAll("Date=.*", "#####DATE");
			     responseString = responseString.replaceAll("Set-Cookie=.*", "#####COOKIE");
			     responseString = responseString.replaceAll("CF-RAY=.*", "#####CF-RAY");
		
	
		     return responseString;
		    }

	}
	
	