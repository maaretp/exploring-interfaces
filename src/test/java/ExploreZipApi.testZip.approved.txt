status code: 200
headers: 
Date=Sun, 11 Aug 2019 19:26:14 GMT
Content-Type=application/json
Transfer-Encoding=chunked
Connection=keep-alive
Set-Cookie=__cfduid=d4ac0a437677c3d65a5014f7824ae21081565551574; expires=Mon, 10-Aug-20 19:26:14 GMT; path=/; domain=.zippopotam.us; HttpOnly
X-Cache=hit
Charset=UTF-8
Vary=Accept-Encoding
Access-Control-Allow-Origin=*
Server=cloudflare
CF-RAY=504c999ac9ede0ce-IAD
Content-Encoding=gzip
body: 
{
    "post code": "20500",
    "country": "United States",
    "country abbreviation": "US",
    "places": [
        {
            "place name": "Washington",
            "longitude": "-77.0355",
            "state": "District of Columbia",
            "state abbreviation": "DC",
            "latitude": "38.8946"
        }
    ]
}
